15_mini_project_work/3_magic_ball_game.py

```diff
@@ -7,18 +7,64 @@
            'Перспективы не очень хорошие', 'Весьма сомнительно']
 USER_NAME_PROMPT = 'Введи свое имя:\n'
 EXCLAMATION_SIGN = '!'
+OBLIGATORY_SIGN = '?'
 USER_NAME_GREETING = 'Добро пожаловать, '
 CONTINUE_GAME_PROMPT = 'Хочешь задать еще один вопрос?\nНажми ENTER,' \
                        ' если ДА\n"1", затем ENTER, если НЕТ\n'
 ENTER_QUESTION_PROMPT = 'Введи свой вопрос:\n'
+IGNORE_WORD_LIST = ['где', 'когда', 'почему', 'сколько', 'кто',
+                    'с кем', 'кого', 'зачем', 'почему', 'чего',
+                    'как', 'какой', 'каком', 'какую', 'какие', 'какого']
+IMPOSSIBLE_QUESTION_ERROR = 'Я могу ответить только на вопрос, требующий ответа ДА или НЕТ'
+NON_LETTER_ERROR = 'Вопрос должен содержать хотя бы одну букву или цифру!'
+NO_QUESTION_MARK_ERROR = 'Вопрос должен содержать вопросительный знак!'
 NO_RESPONSE = '1'
 FAREWELL_MESSAGE = 'Возвращайся если возникнут вопросы!'
 
 
+def is_question_valid(question, ignore_word_list, no_question_mark_error, impossible_question_error,
+                      non_letter_error):
+    no_letter = True
+    no_question_mark = True
+    is_question_impossible = False
+
+    for c in question:
+        if c.isalnum():
+            no_letter = False
+        elif c == '?':
```

Comment:
if TRUE -> TRUE 
one-liner needed

Reply:
Does not seem possible, because if >- elif is one-liner non-convertible.